library/ssl_tls13_keys.c

```diff
@@ -654,7 +654,9 @@ int mbedtls_ssl_tls13_calculate_verify_data( mbedtls_ssl_context* ssl,
     unsigned char transcript[MBEDTLS_TLS1_3_MD_MAX_SIZE];
     size_t transcript_len;
 
-    unsigned char const *base_key = NULL;
+    mbedtls_ssl_tls13_handshake_secrets *tls13_hs_secrets =
+                                            &ssl->handshake->tls13_hs_secrets;
+    unsigned char *base_key = tls13_hs_secrets->server_handshake_traffic_secret;
```

Comment:
```suggestion
    unsigned char *base_key = tls13_hs_secrets->server_handshake_traffic_secret;
```
It should rather be:
```C
if( from == MBEDTLS_SSL_IS_CLIENT )
    base_key = tls13_hs_secrets->client_handshake_traffic_secret;
else
    base_key = tls13_hs_secrets->server_handshake_traffic_secret;
```
right ?